tests/components/zha/test_light.py

```diff
@@ -449,119 +474,180 @@ async def async_test_zha_group_light_entity(
+        hass, group_cluster_identify, group_entity_id, FLASH_SHORT
     )
 
+    # test turning the lights on and off from the light
+    await async_test_on_off_from_light(hass, dev1_cluster_on_off, group_entity_id)
+
     # test turning the lights on and off from the HA
     await async_test_level_on_off_from_hass(
-        hass, group_cluster_on_off, group_cluster_level, entity_id
+        hass, group_cluster_on_off, group_cluster_level, group_entity_id
     )
 
     # test getting a brightness change from the network
-    await async_test_on_from_light(hass, group_cluster_on_off, entity_id)
+    await async_test_on_from_light(hass, dev1_cluster_on_off, group_entity_id)
     await async_test_dimmer_from_light(
-        hass, group_cluster_level, entity_id, 150, STATE_ON
+        hass, dev1_cluster_level, group_entity_id, 150, STATE_ON
     )
 
     # test long flashing the lights from the HA
     await async_test_flash_from_hass(
-        hass, group_cluster_identify, entity_id, FLASH_LONG
+        hass, group_cluster_identify, group_entity_id, FLASH_LONG
     )
 
+    assert len(zha_group.members) == 2
     # test some of the group logic to make sure we key off states correctly
-    await dev1_cluster_on_off.on()
-    await dev2_cluster_on_off.on()
+    await send_attributes_report(hass, dev1_cluster_on_off, {0: 1})
+    await send_attributes_report(hass, dev2_cluster_on_off, {0: 1})
+    await hass.async_block_till_done()
 
     # test that group light is on
-    assert hass.states.get(entity_id).state == STATE_ON
+    assert hass.states.get(device_1_entity_id).state == STATE_ON
+    assert hass.states.get(device_2_entity_id).state == STATE_ON
+    assert hass.states.get(group_entity_id).state == STATE_ON
 
-    await dev1_cluster_on_off.off()
+    await send_attributes_report(hass, dev1_cluster_on_off, {0: 0})
+    await hass.async_block_till_done()
 
     # test that group light is still on
-    assert hass.states.get(entity_id).state == STATE_ON
+    assert hass.states.get(device_1_entity_id).state == STATE_OFF
+    assert hass.states.get(device_2_entity_id).state == STATE_ON
+    assert hass.states.get(group_entity_id).state == STATE_ON
 
-    await dev2_cluster_on_off.off()
+    await send_attributes_report(hass, dev2_cluster_on_off, {0: 0})
+    await hass.async_block_till_done()
 
     # test that group light is now off
-    assert hass.states.get(entity_id).state == STATE_OFF
+    assert hass.states.get(device_1_entity_id).state == STATE_OFF
+    assert hass.states.get(device_2_entity_id).state == STATE_OFF
+    assert hass.states.get(group_entity_id).state == STATE_OFF
 
-    await dev1_cluster_on_off.on()
+    await send_attributes_report(hass, dev1_cluster_on_off, {0: 1})
+    await hass.async_block_till_done()
 
     # test that group light is now back on
-    assert hass.states.get(entity_id).state == STATE_ON
+    assert hass.states.get(device_1_entity_id).state == STATE_ON
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)